### PR TITLE
Migrate to MoviebaseApp/trakt-api

### LINKED ISSUE #1155: Remove threetenabp library
Remove the ThreeTen library after migration of MoviebaseApp/trakt-api in #1154

### PR DESCRIPTION
**Pull request by chrisbanes**
Started by @chrisnkrueger in #1154, but moving it to the main repo.

Initial try to integrate the trakt-api. There is some more work:

- [x] Testing the trakt-api requests and reviewing the API
- [x] Test the ZonedDateTime in Kotlin Datetime library (never looked into the parsing)
- [x] Authentication / login with trakt
- [x] Fix the unit tests
- [x] Improve module dependency, currently trakt-auth -> trakt-api ->trakt-auth- android
- [x] Test login. It works but only after restarting the app after logging in. I guess that the Ktor needs to re-retrieve the auth token it is using.
- [x] Fix marking episodes as watched.

Related to #1091. Closes #1155

### CODE REVIEW

**Comment by chrisnkrueger** [comment]
> kotlinx datetime doesn't contain a ZonedDateTime type. Luckily for us, all responses from Trakt sent in UTC ([here](https://trakt.docs.apiary.io/#introduction/dates)) so we can just use Instant

Sorry for the confusion. Kotlin Datetime uses `OffsetDateTime` under the hood, so we are okay with the JSON  format "2020-12-09T09:16:56.000124Z" (`InstantIso8601Serializer`).

File: gradle/libs.versions.toml [hunk 1]
```diff
@@ -147,8 +146,6 @@ swipe = "me.saket.swipe:swipe:1.1.1"
 
 store = "org.mobilenativefoundation.store:store5:5.0.0-alpha03"
 
-threeTenAbp = "com.jakewharton.threetenabp:threetenabp:1.4.4"
```
Comment (chrisnkrueger) [inline gradle/libs.versions.toml:154]: Nice +1

**Comment by chrisbanes** [comment]
The last commits needs https://github.com/MoviebaseApp/trakt-api/pull/16 and https://github.com/MoviebaseApp/tmdb-api/pull/53 to be deployed before it will build.

File: data/legacy/src/main/java/app/tivi/data/mappers/TraktStatusToShowStatus.kt [hunk 2]
```diff
@@ -16,16 +16,18 @@
...
 @Inject
-class TraktStatusToShowStatus() : Mapper<Status, ShowStatus> {
-    override suspend fun map(from: Status) = when (from) {
-        Status.ENDED -> ShowStatus.ENDED
-        Status.RETURNING -> ShowStatus.RETURNING
-        Status.CANCELED -> ShowStatus.CANCELED
-        Status.IN_PRODUCTION -> ShowStatus.IN_PRODUCTION
+class TraktStatusToShowStatus() : Mapper<TraktShowStatus, ShowStatus> {
+
+    override suspend fun map(from: TraktShowStatus) = when (from) {
```
Comment (chrisnkrueger) [inline data/legacy/src/main/java/app/tivi/data/mappers/TraktStatusToShowStatus.kt:26]: > Note: When getting full extended info, the status field can have a value of returning series (airing right now), continuing (airing right now), in production (airing soon), planned (in development), upcoming (in development), pilot, canceled, or ended.

https://trakt.docs.apiary.io/#reference/shows/summary

Comment (chrisbanes) [reply data/legacy/src/main/java/app/tivi/data/mappers/TraktStatusToShowStatus.kt]: Thanks for the pointer!

**Comment by chrisnkrueger** [comment]
🎉

**Pull request merged**